vm/src/lib.rs

```diff
@@ -111,6 +118,46 @@ impl VM {
         Ok(vm)
     }
 
+    /// Sets the activation height for a specific host query.
+    ///
+    /// This method associates a previously registered host query with a block
+    /// height at which it becomes active. Before this activation height,
+    /// the host query will be excluded from session execution.
+    ///
+    /// **Note:** The specified host query must already be registered in the
+    /// global host queries registry before calling this method.
+    ///
+    /// # Arguments
+    /// * `host_query` - The name of the host query to activate.
+    /// * `activation` - The block height at which the host query becomes
+    ///   active.
+    ///
+    /// # Panics
+    /// This method will panic if the provided `host_query` is not already
+    /// registered in the global host queries registry.
+    ///
+    /// # Examples
+    /// ```rust
+    /// use dusk_vm::VM;
+    /// use dusk_core::abi::Query;
+    ///
+    /// let mut vm = VM::ephemeral().unwrap();
+    /// vm.with_hq_activation(Query::KECCAK256, 100);
```

Comment:
Do we keep the keccak host-query? Because if not it might make sense to use a different example host-query here. Maybe a similar exotic one like the `Query::VERIFY_GROTH16_BN254`?

Reply:
Afaik it's needed by @milosz. Or not?